src/Dsll.hs

```diff
@@ -104,42 +122,24 @@ instance Show Pd2 where
               bString = show propsPd2
               cString = show toPd2
 
-
--- | DSLL program composed out of type constructors followed by ar declarations, operations and predicates
-data DSLLProg = DSLLProg { cd :: [Cd],
```

Comment:
I would still store DSLL programs in this data structure (easier to analyze). I'd just loosen up the parsing order.

Reply:
Where do we need to analyse it (further than the typechceker which does not use it at all)?

Reply:
ok, I admit we aren't analyzing it anywhere, but I think it's neater to have all decls of the same type grouped together in the datatype :-)